serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -49,147 +53,171 @@
     private ELMALookupService elmaLookupService;
     private EntityService entityService;
     private SvarUtService svarUtService;
+    private ProcessService processService;
+    private DocumentTypeService documentTypeService;
+    private SRRequestScope requestScope;
     private static final String NORWAY_PREFIX = "9908:";
 
     /**
      * Creates factory to create ServiceRecord using provided environment and services
-     *
-     * @param properties - parameters needed to contact the provided services
-     * @param virksertService - used to lookup virksomhetssertifikat (certificate)
+     * @param properties        - parameters needed to contact the provided services
+     * @param virksertService   - used to lookup virksomhetssertifikat (certificate)
      * @param elmaLookupService - used to lookup hostname of Altinn formidlingstjeneste
-     * @param krrService - used to lookup parameters needed to use DPI transportation
+     * @param krrService        - used to lookup parameters needed to use DPI transportation
+     * @param entityService     - used to look up information about citizens and organizations in Brønnøysundregisteret and Datahotellet.
+     * @param svarUtService     - used to determine whether an organization utilizes FIKS.
+     * @param documentTypeService
+     * @param requestScope
      */
-    @Autowired
     public ServiceRecordFactory(ServiceregistryProperties properties,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)